test/integration/commands/flex/flex-init.test.js

```diff
@@ -0,0 +1,288 @@
+
+          runSupposeSequence(sequence, (error, exitCode) => {
+            assertions.assertSuccessfulFlexInitSequence(error, exitCode, expectedAppProject, outputFile, flexInitSuccessMessage, done);
+          });
+        }, done);
+      });
+    });
+
+    it('from two profiles should use the active one', (done) => {
+      setup.createProfiles(secondValidProfileName, () => {
+        setup.setActiveProfile(defaultProfileName, true, () => {
+          setup.startMockServer(null, (server) => {
+            ms = server;
+            const sequence = buildSupposeFlexInitSequence(defaultFlexInitOptions, defaultEnvWithDebug);
+
+            runSupposeSequence(sequence, (error, exitCode) => {
+              assertions.assertSuccessfulFlexInitSequence(error, exitCode, expectedAppProject, outputFile, flexInitSuccessMessage, done);
+            });
+          }, done);
+        });
+      });
+    });
+
+    it('should use the submitted profile as an option', (done) => {
+      setup.createProfiles(defaultProfileName, () => {
+        setup.setActiveProfile(secondValidProfileName, true, () => {
+          setup.startMockServer(null, (server) => {
+            ms = server;
+            const profileOptions = buildProfileOptions(defaultFlexInitOptions, defaultProfileName);
+            const sequence = buildSupposeFlexInitSequence(profileOptions, defaultEnvWithDebug);
+
+            runSupposeSequence(sequence, (error, exitCode) => {
+              assertions.assertSuccessfulFlexInitSequence(error, exitCode, expectedAppProject, outputFile, flexInitSuccessMessage, done);
+            });
+          }, done);
+        });
+      });
+    });
+
+    it('should use the submitted profile as an environment variable', (done) => {
+      const envWithProfileVar = {
+        env: cloneDeep(defaultEnv),
+        debug: fs.createWriteStream(outputFile)
+      };
+      envWithProfileVar.env[profileEnvVarName] = defaultProfileName;
+      setup.createProfiles(defaultProfileName, () => {
+        setup.setActiveProfile(secondValidProfileName, true, () => {
+          setup.startMockServer(null, (server) => {
+            ms = server;
+            const sequence = buildSupposeFlexInitSequence(defaultFlexInitOptions, envWithProfileVar);
+
+            runSupposeSequence(sequence, (error, exitCode) => {
+              assertions.assertSuccessfulFlexInitSequence(error, exitCode, expectedAppProject, outputFile, flexInitSuccessMessage, done);
+            });
+          }, done);
+        });
+      });
+    });
+
+    it('submitted profile as an option should be used if there is another profile set as an environment variable', (done) => {
+      const envWithProfileVar = {
+        env: cloneDeep(defaultEnv),
+        debug: fs.createWriteStream(outputFile)
+      };
+      envWithProfileVar.env[profileEnvVarName] = secondValidProfileName;
+      setup.createProfiles(defaultProfileName, () => {
+        setup.setActiveProfile(secondValidProfileName, true, () => {
```

Comment:
If an error occurs here?

Reply:
By the way, I think that at least for some of the tests the profile creation step can be extracted to a `before` hook.